runtimes/go/storage/objects/internal/providers/s3/bucket.go

```diff
@@ -46,8 +48,9 @@ func (mgr *Manager) Matches(cfg *config.BucketProvider) bool {
 func (mgr *Manager) NewBucket(provider *config.BucketProvider, runtimeCfg *config.Bucket) types.BucketImpl {
 	client := mgr.clientForProvider(provider)
 	return &bucket{
-		client: client,
-		cfg:    runtimeCfg,
+		client:         client,
+		presign_client: s3.NewPresignClient(client),
```

Comment:
Instantiate in `clientForProvider` instead so we can cache it

Reply:
Done, thanks